SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -162,7 +163,25 @@ public SingularityRequestParent bounce(@ApiParam("The request ID to bounce") @Pa
 
     SlavePlacement placement = requestWithState.getRequest().getSlavePlacement().or(configuration.getDefaultSlavePlacement());
 
-    final boolean isIncrementalBounce = bounceRequest.isPresent() && bounceRequest.get().getIncremental().or(false);
+    SingularityBounceRequest bounceRequest;
+
+    if (maybeBounceRequest.isPresent()) {
+      bounceRequest = new SingularityBounceRequest(
+        maybeBounceRequest.get().getIncremental(),
+        maybeBounceRequest.get().getSkipHealthchecks(),
+        maybeBounceRequest.get().getDurationMillis().or(Optional.of(TimeUnit.MINUTES.toMillis(configuration.getDefaultBounceExpirationMinutes()))),
+        maybeBounceRequest.get().getActionId().or(Optional.of(UUID.randomUUID().toString())),
+        maybeBounceRequest.get().getMessage());
+    } else {
+      bounceRequest = new SingularityBounceRequest(
+        Optional.<Boolean>absent(),
+        Optional.<Boolean>absent(),
+        Optional.of(TimeUnit.MINUTES.toMillis(configuration.getDefaultBounceExpirationMinutes())),
```

Comment:
Instead of storing this in the object, why not check it later? Recreating the object like this is gross.